File: 0145.binary_tree_postorder_traversal/solution.py
from typing import List

import utils


class Solution:
    def postorder_traversal(self, root: utils.TreeNode) -> List[int]:
        def dfs(node, ans):
            if not node:
                return ans
            dfs(node.left, ans)
            dfs(node.right, ans)
            ans.append(node.val)
            return ans

        return dfs(root, [])

File: 0145.binary_tree_postorder_traversal/solution.go
package leetcode0145

import "github.com/Ackerr/Algorithms/utils"

func postorderTraversal(root *utils.TreeNode) []int {
	return dfs(root, []int{})
}

func dfs(node *utils.TreeNode, ans []int) []int {
	if node == nil {
		return ans
	}
	ans = dfs(node.Left, ans)
	ans = dfs(node.Right, ans)
	ans = append(ans, node.Val)
	return ans
}
